recipe/meta.yaml

```diff
@@ -305,27 +288,28 @@ outputs:
         - typing_extensions
         - {{ pin_subpackage('libtorch', exact=True) }}
       run:
+        - {{ pin_subpackage('libtorch', exact=True) }}
```

Comment:
The old syntax was to workaround the fact that for the non megabuilds each libtorch may have different hashes making it incompatible.

Reply:
Or maybe something like that. Maybe you have addressed the core problem and this is the better way.

Reply:
Even for the megabuild, libtorch has a unique hash, because there's only one. It's pytorch itself that gets different hashes due to the different python versions

Reply:
> Even for the megabuild, libtorch has a unique hash

Not true. See https://anaconda.org/conda-forge/libtorch/files

Reply:
Obviously cuda and non-CUDA create different libtorch hashes, but _within_ a megabuild it's unique, which is what matters for pinning it in pytorch.

Unless the idea was to allow mixing CUDA-enabled pytorch with non-CUDA libtorch, but I don't see the sense in that.

Reply:
It's more interesting for the blas_impl - `pytorch` could theoretically be independent of that (if all the blas calls go through libtorch), but also there we're already creating different `pytorch` hashes there due to the `{{ pin_subpackage("libtorch", exact=True) }}` in the host dependencies, so AFAICT we're not materially changing the various installations here, just making it impossible to install untested/unsupported combinations

Reply:
For non-megabuilds, the idea was to allow libtorch from any of the builds (with the same features except python version) to work with any pytorch build. This way, you don't have to download a different `libtorch` for different python version. Note that this is for non-megabuilds only i.e. osx where we don't have CUDA builds.

Reply:
Yeah, that makes sense! Let me try to reflect that in the run-deps.